src/Microsoft.Framework.Configuration/Constants.cs

```diff
@@ -5,6 +5,6 @@ namespace Microsoft.Framework.Configuration
 {
     public static class Constants
     {
-        public static readonly string KeyDelimiter = ":";
+        public static readonly string[] KeyDelimiter = new[] { ":" };
```

Comment:
@hishamco it is fine to cache the array but I think the code would look cleaner if you kept it separate, e.g. keep `string KeyDelimiter` as is on Constants and add `private static readonly string[] KeyDelimiterArray = new[] { KeyDelimiter }` on ConfigurationKeyCompare to use in that particular case. Seeing the KeyDelimiter[0] on all but that usages is puzzling. Also Constants is public API and you don't want it to look weird because of an optimization like this.